tests/admin_views/tests.py

```diff
@@ -328,6 +328,166 @@ def test_add_with_GET_args(self):
             msg_prefix="Couldn't find an input with the right value in the response",
         )
 
+    def test_add_query_string_persists(self):
+        tests = [
+            {"_addanother": "1"},  # "Save and add another".
+            {"_continue": "1"},  # "Save and continue editing".
+            {"_saveasnew": "1"},  # "Save as new".
+        ]
+        for i, save_option in enumerate(tests):
+            with self.subTest(save_option):
+                url = reverse("admin:auth_user_add")
+                response = self.client.post(
+                    f"{url}?username=newuser",
+                    {
+                        "username": f"newuser{i}",
+                        "password1": "newpassword",
+                        "password2": "newpassword",
+                        **save_option,
+                    },
+                )
+                self.assertIn("username=newuser", response.url)
+
+    def test_add_query_string_persists_with_changelist_filters(self):
+        tests = [
+            {"_addanother": "1"},  # "Save and add another".
+            {"_continue": "1"},  # "Save and continue editing".
+            {"_saveasnew": "1"},  # "Save as new".
+        ]
+        for i, save_option in enumerate(tests):
+            with self.subTest(save_option):
+                url = reverse("admin:auth_user_add")
+                response = self.client.post(
+                    f"{url}?_changelist_filters=is_staff__exact%3D0",
```

Comment:
We should check that currently persisted options work together with other query strings, so I'd add `username=newuser` to new tests:
```suggestion
                    f"{url}?username=newuser&_changelist_filters=is_staff__exact%3D0",
```